src/renate/cli/parsing_functions.py

```diff
@@ -493,32 +638,96 @@ def get_transforms_kwargs(
+    """Returns all argument names of the function and appends new arguments to ``all_args``.
+
+    Args:
+        config_module: Renate config file containing functions to access the model and data.
+        function_name: Function of which we want to return the arguments.
+        all_args: List of all arguments that will be an input to the script.
+        ignore_args: Arguments which will not be added to ``all_args`` because these values will
+            be created on the fly and are no input to the script.
+    """
+    if not hasattr(config_module, function_name):
+        return []
+    arg_spec = inspect.getfullargspec(getattr(config_module, function_name))
+    known_args = [arg for arg in arg_spec.args if arg in all_args]
+    new_args = [arg for arg in arg_spec.args if arg not in all_args and arg not in ignore_args]
+    if arg_spec.defaults is None:
+        default_values = {}
+    else:
+        default_values = {
+            arg_spec.args[len(arg_spec.args) - len(arg_spec.defaults) + i]: default
+            for i, default in enumerate(arg_spec.defaults)
+        }
+    for argument_name in known_args:
+        argument_type = get_argument_type(arg_spec, argument_name)
+        expected_type = all_args[argument_name]["type"]
+        if argument_type != expected_type:
+            raise TypeError(
+                f"Types of `{argument_name}` are not consistent. Defined as type `{argument_type}`"
+                f" as well as `{expected_type}`."
+            )
+        if argument_name not in default_values:
+            all_args[argument_name]["required"] = True
+
+    for argument_name in new_args:
+        all_args[argument_name] = {
+            "type": get_argument_type(arg_spec, argument_name),
+            "argument_group": CUSTOM_ARGS_GROUP,
+        }
+        if argument_name in default_values:
+            all_args[argument_name]["default"] = default_values[argument_name]
+        else:
+            all_args[argument_name]["required"] = True
+    return arg_spec.args
 
 
 parse_by_updater = {
-    "ER": parse_experience_replay_arguments,
-    "DER": parse_dark_experience_replay_arguments,
-    "POD-ER": parse_pod_experience_replay_arguments,
-    "CLS-ER": parse_cls_experience_replay_arguments,
-    "Super-ER": parse_super_experience_replay_arguments,
-    "GDumb": parse_gdumb_arguments,
-    "Joint": parse_joint_arguments,
-    "FineTuning": parse_finetuning_arguments,
-    "RD": parse_rd_learner_arguments,
-    "OfflineER": parse_replay_learner_arguments,
+    "ER": _add_experience_replay_arguments,
+    "DER": _add_dark_experience_replay_arguments,
+    "POD-ER": _add_pod_experience_replay_arguments,
+    "CLS-ER": _add_cls_experience_replay_arguments,
+    "Super-ER": _add_super_experience_replay_arguments,
+    "GDumb": _add_gdumb_arguments,
+    "Joint": _add_joint_arguments,
+    "FineTuning": _add_finetuning_arguments,
+    "RD": _add_rd_learner_arguments,
+    "OfflineER": _add_replay_learner_arguments,
```

Comment:
Do we really need to maintain these functions? can we remove them in favor of an automatic detection of the arguments of each MU with get_function_args (or similar)?

Reply:
These functions extend the arguments parsed by argsparse. I don't see how we can remove this.

Reply:
As discussed offline, let's open a separate issue for this